internal/cli/job_cancel.go

```diff
@@ -59,14 +60,26 @@ func (c *JobCancelCommand) Run(args []string) int {
 		return 1
 	}
 
-	s.Update("Marked job %q for cancellation", jobId)
+	if !c.flagForce {
+		s.Update("Marked job %q for cancellation", jobId)
+	} else {
+		s.Update("Forcefully marked job %q for cancellation", jobId)
+		s.Status(terminal.StatusWarn)
+	}
 	s.Done()
 
 	return 0
 }
 
 func (c *JobCancelCommand) Flags() *flag.Sets {
 	return c.flagSet(flagSetOperation, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    "force",
+			Target:  &c.flagForce,
+			Default: false,
+			Usage:   "Will attempt to forcefully cancel a job.",
```

Comment:
I would recommend adding a lot more warning help text here. Because **force cancel is very dangerous.** I didn't expose it on purpose because people tend to just give it a shot if cancel doesn't work right away, but it is _very dangerous_. I think it makes sense to expose it but the help text should reflect the danger.

A forcefully canceled job will attempt to send a cancellation signal but it is not guaranteed. If the job is actually still running, the result will never be read. The status of it will never be followed up. If it is stuck, it will remain stuck forever. If another job was queued for the same app/workspace, it will begin running immediately which can cause concurrency issues depending on the plugin. 

A force cancel will _never fail_ to update the server state, but _may fail_ to clean up resources. A graceful cancel will _never fail_ to clean up resources, but _may fail_ to cancel the server state (since it gracefully waits).

My recommendation would perhaps be: "Will forcefully cancel the job. This will immediately mark the job as complete in the server, regardless of the real job status. This may leave dangling resources or cause concurrency issues if the underlying job doesn't gracefully cancel."

Note I specifically removed "attempt" from the help text. Force cancel cannot fail.

Reply:
Roger that. Good call! I updated this PR to include some warning text, additional CLI UX warnings on if the flag is used, and the CLI will sleep for a few seconds to give users a chance to ctrl-c if they aren't sure about force cancel.